pkg/ingester/flush.go

```diff
@@ -158,6 +163,18 @@ func (i *Ingester) collectChunksToFlush(instance *instance, fp model.Fingerprint
 		return nil, nil
 	}
 
+	// Flush a chunk after it received no samples for a long time.
+	if len(stream.chunks) == 1 {
```

Comment:
Not keen on the fact that the code handled 1 and >1 chunks differently.  Maybe a for loop?  Use Cortex for inspiration.

```golang
  var result []*chunkDesc
  for _, chunk := range stream.chunks {
    if immediate || shouldFlushChunk(chunk) {
      chunk.closed = true
      result = append(result, chunk)
    }
  }
  return result
}

func shouldFlushChunk(chunk) bool {
  // Append should close the chunk when the a new one is added.
  if chunk.Closed {
    return true
  } 

  if time.Since(chunk.lastUpdated) > i.cfg.MaxChunkIdle {
    return true
  }

  return false
}
```